src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -736,6 +778,58 @@ private bool ArePointerTypesEqual(PointerTypeSymbol type, PointerTypeSymbol othe
                 return AreTypesEqual(type.PointedAtType, other.PointedAtType);
             }
 
+            private bool AreFunctionPointerTypesEqual(FunctionPointerTypeSymbol type, FunctionPointerTypeSymbol other)
+            {
+                var sig = type.Signature;
+                var otherSig = other.Signature;
+
+                ValidateFunctionPointerParamOrReturn(
+                    sig.ReturnTypeWithAnnotations, otherSig.ReturnTypeWithAnnotations,
+                    sig.RefKind, otherSig.RefKind,
+                    sig.RefCustomModifiers, otherSig.RefCustomModifiers,
+                    allowOut: false);
+                if (sig.RefKind != otherSig.RefKind || !AreTypesEqual(sig.ReturnTypeWithAnnotations, otherSig.ReturnTypeWithAnnotations))
+                {
+                    return false;
+                }
+
+                return sig.Parameters.SequenceEqual(otherSig.Parameters, AreFunctionPointerParametersEqual);
+            }
+
+            private bool AreFunctionPointerParametersEqual(ParameterSymbol param, ParameterSymbol otherParam)
+            {
+                ValidateFunctionPointerParamOrReturn(
+                    param.TypeWithAnnotations, otherParam.TypeWithAnnotations,
+                    param.RefKind, otherParam.RefKind,
+                    param.RefCustomModifiers, otherParam.RefCustomModifiers,
+                    allowOut: false);
+
+                return param.RefKind == otherParam.RefKind && AreTypesEqual(param.TypeWithAnnotations, otherParam.TypeWithAnnotations);
+            }
+
+            [Conditional("DEBUG")]
+            private void ValidateFunctionPointerParamOrReturn(TypeWithAnnotations type, TypeWithAnnotations otherType, RefKind refKind, RefKind otherRefKind, ImmutableArray<CustomModifier> refCustomModifiers, ImmutableArray<CustomModifier> otherRefCustomModifiers, bool allowOut)
+            {
+
+                Debug.Assert(type.CustomModifiers.IsEmpty);
+                Debug.Assert(otherType.CustomModifiers.IsEmpty);
+                Debug.Assert(verifyRefModifiers(refCustomModifiers, refKind, allowOut));
+                Debug.Assert(verifyRefModifiers(otherRefCustomModifiers, otherRefKind, allowOut));
+
+                static bool verifyRefModifiers(ImmutableArray<CustomModifier> modifiers, RefKind refKind, bool allowOut)
+                {
+                    Debug.Assert(RefKind.RefReadOnly == RefKind.In);
+                    switch (refKind)
+                    {
+                        case RefKind.RefReadOnly:
+                        case RefKind.Out when allowOut:
```

Comment:
>allowOut [](start = 46, length = 8)

When is `allowOut` set to `true`?

Reply:
Copy/paste error, would have been caught by a test failure.

---
In reply to: [425363880](https://github.com/dotnet/roslyn/pull/44256#discussion_r425363880) [](ancestors = 425363880)